File: Exemplos M10/App_IntentDigitalHUB_Ionic/Elgin_IntentDigitalHUB_Ionic-font/src/intentDigitalHubService/SAT/Commands/AssociarAssinatura.ts
import { SatCommand } from "../SatCommand";

export class AssociarAssinatura extends SatCommand {
  readonly numSessao: number;
  readonly codAtivacao: string;
  readonly cnpjSH: string;
  readonly assinaturaAC: string;

  constructor(
    numSessao: number,
    codAtivacao: string,
    cnpjSH: string,
    assinaturaAC: string
  ) {
    super("AssociarAssinatura");
    this.numSessao = numSessao;
    this.codAtivacao = codAtivacao;
    this.cnpjSH = cnpjSH;
    this.assinaturaAC = assinaturaAC;
  }

  functionParameters(): object {
    return {
      numSessao: this.numSessao,
      codAtivacao: this.codAtivacao,
      cnpjSH: this.cnpjSH,
      assinaturaAC: this.assinaturaAC,
    };
  }
}

File: Exemplos M10/App_IntentDigitalHUB_XamarinAndroid/Elgin_IntentDigitalHUB_XamarinAndroid/IntentServices/Sat/AssociarAssinatura.cs
namespace Xamarin_Android_Intent_Digital_Hub.IntentServices.Sat
{
    class AssociarAssinatura : SatCommand
    {
        readonly int numSessao;
        readonly string codAtivacao;
        readonly string cnpjSH;
        readonly string assinaturaAC;

        public AssociarAssinatura(int numSessao, string codAtivacao, string cnpjSH, string assinaturaAC) : base("AssociarAssinatura")
        {
            this.numSessao = numSessao;
            this.codAtivacao = codAtivacao;
            this.cnpjSH = cnpjSH;
            this.assinaturaAC = assinaturaAC;
        }

        protected override string FunctionParameters()
        {
            return "\"numSessao\"" + ":" + numSessao + "," +
                    "\"codAtivacao\"" + ":" + "\"" + codAtivacao + "\"" + "," +
                    "\"cnpjSH\"" + ":" + "\"" + cnpjSH + "\"" + "," +
                    "\"assinaturaAC\"" + ":" + "\"" + assinaturaAC + "\"";
        }
    }
}